mlir/test/Target/Cpp/common-cpp.mlir

```diff
@@ -85,8 +85,12 @@ func.func @opaque_types(%arg0: !emitc.opaque<"bool">, %arg1: !emitc.opaque<"char
 func.func @apply(%arg0: i32) -> !emitc.ptr<i32> {
   // CHECK: int32_t* [[V2]] = &[[V1]];
   %0 = emitc.apply "&"(%arg0) : (i32) -> !emitc.ptr<i32>
```

Comment:
I think we should only allow applying address-of to lvalues, i.e.
`%0 = emitc.apply "&"(%arg0) : (!emitc.lvalue<i32>) -> !emitc.ptr<i32>`
C only allows it on lvalues, too.

Reply:
Agreed

Reply:
Done